sympy/calculus/tests/test_hamilton.py

```diff
@@ -0,0 +1,84 @@
+from sympy import Symbol, Function, Derivative as D, Eq, cos, sin
+from sympy.utilities.pytest import raises
+from sympy.calculus.hamilton import hamilton_equations as hamilton
+
+
+def test_hamilton_error1():
+    # Test by giving the Hamiltonian Interface bad-functions
+    x = Function('x')
+    p = Function('p')
+    t = Symbol('t')
+    h1 = x(t)**2 / 2 + p(t)**2 / 2
+    h2 = x(p)**2 / 2 + p(t)**2 / 2
+    raises(TypeError, lambda: hamilton(h1, x(t), p, t))
+    raises(TypeError, lambda: hamilton(h2, x(t), p(t), t))
+
+
+def test_hamilton_error2():
+    x = Function('x')
+    p = Function('p')
+    t = Symbol('t')
+    h = x(t) ** 2 / 2 + p(t) ** 2 / 2
+    # Giving function instead of symbol as time
+    raises(TypeError, lambda: hamilton(h, [x(t)], [p(t)], x(t)))
```

Comment:
These tests are quite repetitive. They should probably be combined into a single `test_` function.

Reply:
I merged the Error cases, but not the SHM because soon I want to write better tests for more complex systems and it's easier to visualise each system in it's own function rather than clubbed together.